### ISSUE TITLE
Usefulness of "Application Name"

### ISSUE THREAD

**Issue by FireInWinter** [mysql-net/MySqlConnector#587]
I've been playing with using the ApplicationName in the connection string, but it seems to be of very limited value.  In my testing of MySqlConnector 0.47.1 the entry only exists in the session_connect_attrs table on new connections.  If the connection is reset and put in the connection pool, the entry (actually all entries) are cleared from session_connect_attrs for the session.  I've testing this with the Connector/Net driver (for the default entries it creates, not application name) and their entries are removed as well.

Since this setting seems to only work in some scenarios, is it worth having it at all?  If you use connection pooling (the default) with the connections being reset (also the default), it seems you will hardly ever be able to see the entry in the session_connect_attrs table.  At the very least I think the documentation for the connection options should be updated to state when it will work.

I tested both Connector/Net and MySqlConnector with similar connection strings.
```
Connector/Net
server=MyServer;user id=MyUser;database=performance_schema;password=MyPassword
```

```
MySqlConnector
server=MyServer;user id=MyUser;database=performance_schema;password=MyPassword;ApplicationName='test'
```
And then used the same query on each to verify it.
```
select * from session_connect_attrs where PROCESSLIST_ID = connection_id()
```
The first time you run the query with the connection string where it resets the connection, you will see entries in session_connect_attrs, but the second time you run it(after closing and opening the connection), it will be empty.  If you either set "Pooling=False" or "connectionreset=False" in the connection string it will work as expected, but that seems to be a very small percentage of cases in modern code.

**Comment by bgrainger** [comment]
Relates to the feature added in #547.

I can't reproduce this with MySQL Server 5.7.21.

MySQL Workbench shows the attribute value:

![image](https://user-images.githubusercontent.com/188129/48870809-12706300-ed97-11e8-9645-fc8f94d380cd.png)

A simple C# program can retrieve the `session_connect_attrs` after resetting the connection:

```csharp
for (var i = 0; i < 5; i++)
{
	using (var connection = new MySqlConnection(csb.ConnectionString))
	{
		connection.Open();
		connection.Query(@"select * from session_connect_attrs where PROCESSLIST_ID = connection_id()").Dump();
	}
}
```

The output shows that the connection is being reset and that the session_connect_attrs are still present:

![image](https://user-images.githubusercontent.com/188129/48870992-c7a31b00-ed97-11e8-862f-57fe9dfb1a10.png)

Finally, I verified that `ConnectionReset=true` was having an effect by creating then selecting rows from a temporary table on two different connections; with `ConnectionReset=true` the temporary table had been destroyed and the program threw an exception.

**Comment by FireInWinter** [comment]
Sorry I didn't include that.  It was 5.6.10 (AWS RDS Aurora) and 5.6.23 (Percona Server).  So if it is just due to something in 5.6, then it wasn't as big a deal as I thought.  I'll just not plan on really have it report much until I upgrade to 5.7, but will keep using it.

**Comment by bgrainger** [comment]
I checked that MySqlConnector is re-sending the connection attributes with the `COM_CHANGE_USER` packet, but MySQL 5.6 doesn't seem to be storing them. I can't say for sure that this is a server-side issue, but because this seems to be resolved with MySQL 5.7 (and later), I'm not planning to look into this further.
